src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1448,6 +1448,10 @@ private async Task ReconnectAsync(Exception? closeException)
                 nextRetryDelay = GetNextRetryDelay(previousReconnectAttempts++, DateTime.UtcNow - reconnectStartTime, retryReason);
             }
 
+            // we incremented then saw that the next reconnect delay was null, meaning reconnect attempts are exhausted,
+            // so decrement to have the correct number for logs and exceptions
+            previousReconnectAttempts--;
```

Comment:
Should wee just increment this at the start of the `while (nextRetryDelay != null)` so we don't have to decrement? Or maybe after the `Task.Delay`, since if it was canceled by shutdown or something we technically didn't attempt a reconnect.

Reply:
So I changed it to try that out, it required changing a couple other spots too. Take a quick look to make sure it's not obviously wrong, or if we prefer the first way.